Sources/SwiftSyntaxBuilder/FunctionCallExprConvenienceInitializers.swift

```diff
@@ -14,21 +14,26 @@
 import SwiftSyntax
 
 extension FunctionCallExpr {
+  /// A convenience initializer that allows passing in arguments using a result builder
+  /// instead of having to wrap them in a `TupleExprElementList`.
+  /// The presence of the parenthesis will be inferred based on the presence of arguments and trailing closures.
   public init(
     _ calledExpression: ExpressibleAsExprBuildable,
-    leftParen: TokenSyntax? = .leftParen,
-    rightParen: TokenSyntax? = .rightParen,
     trailingClosure: ExpressibleAsClosureExpr? = nil,
     @TupleExprElementListBuilder argumentListBuilder: () -> ExpressibleAsTupleExprElementList = { TupleExprElementList([]) },
     @MultipleTrailingClosureElementListBuilder additionalTrailingClosuresBuilder: () -> MultipleTrailingClosureElementList? = { nil }
   ) {
+    let additionalTrailingClosures = additionalTrailingClosuresBuilder()
+    let argumentList = argumentListBuilder().createTupleExprElementList()
+    let shouldEmitParens = argumentList.elements.isEmpty == false
+      || (trailingClosure == nil && additionalTrailingClosures?.elements.isEmpty != false)
```

Comment:
A couple of suggestions here that improve legibility IMHO.
- I found this easier to read if it‘s storing whether to omit parenthesis rather than when to emit them. I think this is because I’m mentally thinking about the edge case where they are not necessary rather than the normal cases where they are required
- Since for valid code having additional trailing closures implies having a “normal” trailing closure, I don’t think we need to check for `additionalTrailingClosures?.elements.isEmpty` here

```suggestion
    let shouldOmitParens = argumentList.elements.isEmpty && trailingClosure != nil
```